tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -1,90 +1,279 @@
-// - instance exist
-// - in running state
-// - has public subnet
-// - Public IP
-// - OpenShift cluster Worker SG
-// - Windows SG
-// - OpenShift cluster IAM
-// - has name
-// - has OpenShift cluster vpc
-// - instance and SG IDs are recorded to the json file
-// After termination, it checks for the following:
-// - instance terminated
-// - Windows SG deleted
-// - json file deleted
-func TestCreateDestroyWindowsInstance(t *testing.T) {
+var (
 	// Get kubeconfig, AWS credentials, and artifact dir from environment variable set by the OpenShift CI operator.
-	kubeconfig := os.Getenv("KUBECONFIG")
-	awscredentials := os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
-	dir := os.Getenv("ARTIFACT_DIR")
+	kubeconfig     = os.Getenv("KUBECONFIG")
+	awscredentials = os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+	dir            = os.Getenv("ARTIFACT_DIR")
 
-	// The e2e test uses Microsoft Windows Server 2019 Base with Containers image, m4.large type, and libra ssh key.
 	// The CI-operator uses AWS region `us-east-1` which has the corresponding image ID: ami-0b8d82dea356226d3 for
 	// Microsoft Windows Server 2019 Base with Containers.
-	cloud, err := cloudprovider.CloudProviderFactory(kubeconfig, awscredentials, "default", dir,
-		"ami-0b8d82dea356226d3", "m4.large", "libra")
-	assert.NoError(t, err, "error creating clients")
+	imageID      = "ami-0b8d82dea356226d3"
+	instanceType = "m4.large"
+	sshKey       = "libra"
+
+	// awsProvider is setup as a variable for both creating, destroying,
+	// and tear down Windows instance in case test fails in the middle.
+	awsProvider = &awscp.AwsProvider{}
+
+	// Set global variables for instance object, instance, security group,
+	// and infrastructure IDs so that once they are created,
+	// they will be used by all subsequent testing functions.
+	createdInstance   = &ec2.Instance{}
+	createdInstanceID = ""
+	createdSgID       = ""
+	infraID           = ""
+)
 
-	err = cloud.CreateWindowsVM()
-	assert.NoError(t, err, "error creating Windows instance")
+// TestAwsE2eSerial runs all e2e tests of AWS implementation in the serial. It creates the Windows instance,
+// checks all properties of the instance and destroy the instance and check that resource are deleted.
+func TestAwsE2eSerial(t *testing.T) {
+	setupAWSCloudProvider(t)
+
+	setupWindowsInstance(t)
+
+	t.Run("test created instance properties", testInstanceProperties)
+	t.Run("test instance is attached a public subnet", testInstanceHasPublicSubnetAndIp)
+	t.Run("test instance has name tag", testInstanceIsAttachedWithName)
+	t.Run("test instance has infrastructure tag", testInstanceInfraTagExists)
+	t.Run("test instance is attached the cluster worker's security group", testInstanceHasClusterWorkerSg)
+	t.Run("test instance is attache a Windows security group", testInstanceIsAssociatedWithWindowsWorkerSg)
+	t.Run("test instance is associated with cluster worker's IAM", testInstanceIsAssociatedWithClusterWorkerIAM)
```

Comment:
I am a nack on this sort of breakdown. You basically have two tests for creation and deletion. These individual "tests" are just asserting that the expected result occurred. It is fine if they are helper functions called from the main test.

Reply:
Oh i see. I will change it into 2 tests.